### PR TITLE
fix: wallet not reloading data after a reconnect and wallet reset after an error

### PR DESCRIPTION
**Pull request by andreabadesso**
## Summary

After the refactor done in https://github.com/HathorNetwork/hathor-wallet-mobile/pull/232, the wallet stopped reloading after a websocket reconnection.

The initial problem was that we were not setting the `walletStartedState` to `LOADING`, which made it reload the wallet but never display the loading screen.

But not only that, since we refactored the loading flow, the reload was still not happening. Upon investigation, I found out that the old mechanism for reloading the wallet was only working because the `LoadHistoryScreen` was dispatching the `startWalletRequested` action on the `componentDidUpdate`, as we can see [here](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/232/files#diff-6b628a954cc3611565a6b19e839717bb5431913b6d02c38206de52df4b355fbbL55).

### Acceptance Criteria
- We should reload the wallet data after a reconnection
- We should refresh the wallet addresses after a reconnection
- We should stop resetting the wallet if an error occurs
- We should display the correct wallet-service connected server while on the wallet-service facade

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by r4mmer — APPROVED** [review]

File: src/sagas/wallet.js [hunk 1]
```diff
@@ -568,16 +567,19 @@ export function* onWalletConnStateUpdate({ payload }) {
...
 
-  // Since we close the channel after a walletReady event is received,
-  // we must fork this saga again so we setup listeners again.
-  yield fork(listenForWalletReady, wallet);
+  // If we are using the wallet-service, we don't need to wait until the addresses
+  // are reloaded since they are stored on the wallet-service itself.
+  if (!useWalletService) {
+    // Since we close the channel after a walletReady event is received,
+    // we must fork this saga again so we setup listeners again.
+    yield fork(listenForWalletReady, wallet);
```
Comment (pedroferreira1) [inline src/sagas/wallet.js:578]: I understand we don't need to wait for the wallet ready to continue if it's wallet service but don't we need this listener anyway for other things, e.g., error handling when the load fails, ws connection lost (this last one I feel it's not affected)?

**Comment by andreabadesso** [comment]
@pedroferreira1 and @r4mmer 

The addresses were not being refreshed after a reload on the wallet-service because we were only refreshing the token balances and history, this was fixed

I had also to change the mechanism for displaying the current address on the `ReceiveMyAddress` component as it was not being updated if the user reconnected while the screen was opened

I also fixed a bug where the wallet would reset if some error was thrown and the user didn't click to try again

File: src/sagas/wallet.js [hunk 2]
```diff
@@ -596,6 +605,14 @@ export function* onWalletReloadData() {
 
       yield put(tokenInvalidateHistory(tokenUid));
     }
+
+    // We also need to refresh the addresses since we are not starting the wallet again
+    yield call(wallet.getNewAddresses.bind(wallet));
```
Comment (pedroferreira1) [inline src/sagas/wallet.js:610]: We don't have this method in the old facade.

Comment (andreabadesso) [reply src/sagas/wallet.js]: Fixed in 4250c651855426d27d8ba3e8c155a2462a4a95b2

Thanks

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**